ci(lint): use `prefer-const` rule from ESLint

overlookmotel opened the pull request:
Dogfood Oxlint JS plugins by enabling ESLint's `prefer-const` rule, running as a JS plugin.

#15707 will implement this rule natively in Oxlint, but in meantime this is a good one to use to test JS plugins.

camc314 commented on oxlint-plugin.mts:
I'm leaning towards the following structure
```
/packages
    /oxc-internal-oxlint-plugin
```

My reasoning is that:
1. `tasks/` is for programs, e.g. something that you invoke that has an output
2. it doesn't fit in `napi` as it's not a `napi` package
3. it doesn't fit in `npm` as it's not an `npm` package
4. it doesn't fit in `apps` as it's not an `app`
5. it doesn't fit in `crates` as it's not a rust crate
6. we can add other `packages` to this as needed for example, i think we'll need one if we want to share code idomatically between parser package/oxlint for raw transfer.

That said... feel free to merge this and do it in a follow up (or i can do it)🙂

overlookmotel replied:
Agreed it doesn't really fit in any of the existing folders.

I don't know about creating a subfolder to contain 1 file! We might want to wait until we've got something else to put in `packages` dir.

It's been a battle getting this working for various weird reasons, so I'm going to merge this as is. I'd prefer to leave the subfolder thing to later. If Boshen goes "what the hell is this packages folder?!", I'd prefer the whole PR didn't get reverted.

overlookmotel committed:
ci(lint): use `prefer-const` rule from ESLint (#16448)

Dogfood Oxlint JS plugins by enabling ESLint's `prefer-const` rule, running as a JS plugin.

#15707 will implement this rule natively in Oxlint, but in meantime this is a good one to use to test JS plugins.
